include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -311,6 +286,36 @@ using __enable_if_device_execution_policy_double_no_default =
                            __is_convertible_to_event<_Events...>,
                        _T>;
 
+template <typename _BackendTag>
+struct __hetero_tag
+{
+    using __backend_tag = _BackendTag;
+};
+
+struct __device_backend_tag
+{
+};
+
+template <class... _IteratorTypes, typename _KernelName>
+::std::enable_if_t<__is_random_access_iterator_v<_IteratorTypes...>, __hetero_tag<__device_backend_tag>>
```

Comment:
Do we want `static_assert` instead of `enable_if`?

Reply:
It's the code from your prototype:
https://github.com/oneapi-src/oneDPL/blob/db90ee7984647d63e8ff49ade3657b9715bcdcf0/include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h#L260

What do you mean?